SI-9745 typedFunction keeps x => f(x)

som-snytt opened the pull request:
To type `x => f(x)`, `typedFunction` uses the expected
type and `f` to type `x`. Previously, it discarded the
result of typechecking `f`, which was a problem in the
case of a `g(e)` where `e` is "erroneous" but rewriteable
to an expr that checks, e.g., `g(i += 1)`.

This commit keeps the rewritten `g(i = i + 1)` instead
of trying to recover the erroneous tree.

Commit:
* restructure to groups new functions in scala.math package

 * bugfixes

zizhengtai committed:
Make SearchImpl a value class to improve performance

som-snytt committed:
SI-9380 Remove Regex.unapplySeq(Any)

Long deprecation cycle and annoying deprecation message.

Changes deprecation to: "error: cannot resolve overloaded unapply".

janekdb committed:
SI-9813: Complete addition of Java 1.6 and 1.8 java.lang.Math methods

Group documentation,

 - Add groups: modquo, adjacent-float and scaling
 - Rename group signum to signs to allow copySigns to be included
 - Add group description to exact group
 - Use single line documentation comment where possible

Parameter names,

 - Revert parameter name changes on existing methods for compatibility
 - Standardize on x if no clearly better alternative

ruippeixotog committed:
SI-9706 Add a fromString method to Numeric trait

l0rinc committed:
Revert temporary duplication of HashMap internals

In 2.12.1, we had to copy/paste a few methods from HashTable to
HashMap to fix a performance issue without breaking binary compatibility.

This commit opens up the access of the methods in the trait and
reuses them in `HashMap.getOrElseUpdate`, and removes the copy
pasted code.

adriaanm committed:
Merge pull request #5568 from retronym/merge/2.12.x-to-2.13.x-20161130

Merge 2.12.x to 2.13.x [ci: last-only]

SethTisue committed:
disable MiMa for now for 2.13

to be brought back sometime around or before 2.13.0-RC1

retronym committed:
Merge pull request #5578 from SethTisue/disable-mima-2.13

disable MiMa for now for 2.13

SethTisue committed:
2.12 -> 2.13 in the most important places

there are a lot of other places that could be updated, especially in
comments/docs, but the sooner we update them all, the more annoying
merge conflicts we'll have to suffer through. so at the moment let's
just hit the main ones

for now, just override it to use 2.12 modules+ScalaCheck. eventually
something will break but for now we'll get by

SethTisue committed:
in spec generation, 2.12 -> 2.13 so we don't collide

SethTisue committed:
Merge pull request #5582 from SethTisue/keep-2.13-spec-separate

in spec generation, 2.12 -> 2.13 so we don't collide

SethTisue committed:
Merge pull request #5581 from SethTisue/twelve-going-on-thirteen

2.12 -> 2.13 in the most important places

xuwei-k committed:
remove unnecessary ClassTag in Sorting.stableSort

lrytz committed:
Merge pull request #5591 from xuwei-k/stableSort-ClassTag

remove unnecessary ClassTag in Sorting.stableSort

monkey-mas committed:
Remove unnecessary methods in TraversableOnce

As the commit(#4939)[1] says, we can remove the following unnecessary methods defined in GenTraversableOnce[2,3,4].

* foreach
* isEmpty
* hasDefiniteSize

[1] https://github.com/scala/scala/commit/f8d6814bb4fb0248d6121335e67eee8fde7ce455
[2] https://github.com/scala/scala/blob/v2.12.1/src/library/scala/collection/GenTraversableOnce.scala#L68
[3] https://github.com/scala/scala/blob/v2.12.1/src/library/scala/collection/GenTraversableOnce.scala#L87
[4] https://github.com/scala/scala/blob/v2.12.1/src/library/scala/collection/GenTraversableOnce.scala#L112

adriaanm committed:
New keypair for 2.13 spec forced command

See .travis.yml for instructions

lrytz committed:
Merge pull request #5548 from janekdb/issue/SI-9813-to-2.13.x

SI-9813: Add Java 1.6. and 1.8 java.lang.Math methods into scala.math.package

lrytz committed:
Merge pull request #5599 from lrytz/pr5342

Make SearchImpl a value class to improve performance

szeiger committed:
Remove parallel collections from scala-library

They will be moved into a separate module as part of the ongoing
modularization of the Scala standard library.

Tests specific to parallel collections will be reinstated in the new
scala-parallel-collections project as JUnit tests or pure ScalaCheck
tests (run directly from sbt).

Some test cases here contain useful tests for both, parallel collections
and other features of the standard library. They are split up into
separate JUnit tests, which leads to some new JUnit tests in this
project, too.

monkey-mas committed:
Change the signature of WrappedArray.newBuilder[T] to return Builder[T, WrappedArray[T]]

As newBuilder methods in the existing API always return a Builder in their public API,
we can return Builder type instead of ArrayBuffer.

The return type has not been changed since it was first added to WrappedArray
in this commit.[1]

[1] https://github.com/scala/scala/commit/d5b02c8652d7edbdfb0b5a02570d370d3bad299f#diff-5999d408677554483bfed5cdce592b1fR73

johnynek committed:
Add substitute methods and tests

som-snytt committed:
SI-10137 Correct Queue.enqueue(collection.Iterable)

The parameter incorrectly progressed to `immutable.Iterable`.

xuwei-k committed:
Make TraversableOnce.MonadOps a value class

sjrd committed:
Remove the error recovery mechanism in RedBlackTree.TreeIterator.

After fixing the stack size computation in
648eba62b0054d7a0442de3f5a79eb75d38e7602, this commit removes the
error recovery mechanism which hid the internal bug. As the
comment in `pushNext` says, "Obviously this should never happen".

This commit is potentially risky, if we still do wrongly compute
the maximum stack size, but further problems are not caught by
the test suite. But at least we would now know when we fail to
compute the stack size correctly, instead of letting it go
unnoticed for years.

This change also ensures that Scala/JVM does not silently succeed
where Scala.js would fail.

fhalde committed:
API to find random number in a range

This commit adds a method named `between` in the
Random class of scala to find a random number
between the range.

adriaanm committed:
Avoid `sys.props`

adriaanm committed:
Avoid `sys.error`

Use `throw new RuntimeException` (or a subclass) instead.

adriaanm committed:
Avoid `sys.exit`

adriaanm committed:
Avoid `sys.process`

SethTisue committed:
remove deprecated stuff from Predef, scala._, scala.runtime

- bye Pair, Triple
- bye conforms
- bye Predef.readLine and friends
- bye DeprecatedConsole
- bye NotNull
- bye RichException
- bye SeqCharSequence
- bye DeprecatedPredef
- bye Responder
- bye ScalaRuntime.hash
- bye runtime.StringFormat

and:
- `final`ed some @deprecatedOverriding things
- removed some @deprecatedNames

some notable things that were *not* touched in this round:
- Manifest/ClassManifest (sigh)
- DelayedInit (sigh)
- StringAdd (tangled w/ reflection, needs re-STARR)

SethTisue committed:
Merge pull request #5638 from som-snytt/issue/10137

SI-10137 Correct Queue.enqueue(collection.Iterable)

adriaanm committed:
Merge pull request #5683 from SethTisue/remove-deprecated-predef-stuff

remove deprecated stuff from Predef, scala._, scala.runtime

szeiger committed:
Eliminate non-matching overloads early for partial function literals

This was already done for function literals. It is important in cases
where only one overload of a method takes a `Function1` or
`PartialFunction`, for example:

    def map[U](f: T => U): R[U]
    def map[U](f: (A, B) => U): R[C]

By recognizing that a partial function literal always conforms to
`PartialFunction[Any, Nothing]`, in a call such as

    m.map { case (a, b) => a }

the non-matching overload (which takes a `Function2`) can be eliminated
early, thus allowing the known types `A` and `B` to be used during
type-checking of the partial function (which would otherwise fail).

Propagation of overloaded function types (originally implemented in
https://github.com/scala/scala/pull/5307) is extended to cover pattern-
matching anonymous functions, too.

janekdb committed:
Reduce scope of val and improve some names

lrytz committed:
Merge pull request #5703 from janekdb/topic/2.13/fix-names-reduce-scope

Reduce scope of val and improve some names

SethTisue committed:
Merge remote-tracking branch 'origin/2.12.x' into merge-2.12.x-onto-2.13.x-20170217

nice clean merge, trivial conflicts only

SethTisue committed:
Merge pull request #5718 from SethTisue/merge-2.12.x-onto-2.13.x-20170217

[ci: last-only] merge 2.12.x onto 2.13.x (Feb 17, 2017)

lrytz committed:
Merge pull request #5720 from sjrd/remove-rbtree-iterator-recovery-mechanism

Remove the error recovery mechanism in RedBlackTree.TreeIterator.

SethTisue committed:
don't use Console.setOut: was deprecated, now deleted

which was causing integrate-bootstrap to fail

fixes https://github.com/scala/scala-dev/issues/315

lrytz committed:
Merge pull request #5738 from SethTisue/sd-315

don't use Console.setOut: was deprecated, now deleted

szeiger committed:
SI-10081 Avoid diverging implicit scope search

Storing empty infos into the infoMap (thus avoiding recomputation)
prevents the endless recursion.

This is an alternative to https://github.com/scala/scala/pull/5734 that
avoids the expensive annotation removal and the regression in
`pos/t1203a.scala`.

SethTisue committed:
make Predef.*CharSequence implicits public again

fixing a recent regression in 2.13

when I was removing deprecated stuff in
https://github.com/scala/scala/pull/5683/
I was going too fast and misunderstood these deprecations.
only the names of the class parameters were supposed to become
private, not the entire class.

note that now that the parameters aren't even vals anymore, they don't
need to have funny `__` names anymore, either.

this was caught by the Scala community build.

adriaanm committed:
Merge pull request #5751 from SethTisue/bring-back-charsequence-implicits

make Predef.*CharSequence implicits public again

SethTisue committed:
Merge remote-tracking branch 'origin/2.12.x' into 2.13.x

only trivial merge conflict involving a method renaming

*   d1d700e15c - (origin/HEAD, origin/2.12.x) Merge pull request #5754 from Philippus/issue/html-tag-in-hover (2 days ago) <Lukas Rytz>
|\
| * cf64718c97 - pattern for entitylink was too narrow, cleaned up the tests (3 days ago) <Philippus Baalman>
* |   f77139532d - Merge pull request #5671 from retronym/topic/stubby-2 (3 days ago) <Lukas Rytz>
|\ \
| * | ad13063058 - Remove non-essential fix for stub symbol failure (4 days ago) <Jason Zaugg>
| * | 7f2d1fab22 - Avoid forcing info transforms of primitive methods (2 weeks ago) <Jason Zaugg>
| * | 37a0eb758e - Avoid stub symbol related crash in backend (2 weeks ago) <Jason Zaugg>
* | |   96a7617eb6 - Merge pull request #5622 from edmundnoble/extra-errs (4 days ago) <Adriaan Moors>
|\ \ \
| * | | 466e52ba58 - Match error lengths (4 weeks ago) <Edmund Noble>
| * | | d1fc98370a - Improved error messages for identically named, differently prefixed types (9 weeks ago) <Edmund Noble>
|  / /
* | |   f2e05c2276 - Merge pull request #5728 from Philippus/issue/html-tag-in-hover (4 days ago) <Lukas Rytz>
|\ \ \
| | |/
| |/|
| * | e3c5df8ec3 - added missing Inline matches to inlineToStr so it is now exhaustive scala.xml.XML.loadString(tag).text will remove all html tags inside the HtmlTag (9 days ago) <Philippus Baalman>
* | |   920bc4e31c - Merge pull request #5743 from som-snytt/issue/10207-bad-update (7 days ago) <Lukas Rytz>
|\ \ \
| * | | 094f7f9463 - SI-10207 Error before update conversion (8 days ago) <Som Snytt>
* | | |   1b4d36f0c4 - Merge pull request #5746 from paulp/pr/partest (7 days ago) <Lukas Rytz>
|\ \ \ \
| |/ / /
|/| | |
| * | | eac00e1a10 - Add partest paths to the list of watched sources. (8 days ago) <Paul Phillips>
|/ / /
* | |   5f1a638cf5 - Merge pull request #5732 from retronym/topic/build-info-malarkey (10 days ago) <Adriaan Moors>
|\ \ \
| * | | 5e9acac8f6 - More predictable performance of SBT build startup, reload (11 days ago) <Jason Zaugg>
|  / /
* | |   759a7b72a1 - Merge pull request #5735 from SethTisue/sd-313 (10 days ago) <Adriaan Moors>
|\ \ \
| * | | ed4ddf5bae - increase timeouts on some sys.process tests (11 days ago) <Seth Tisue>
* | | |   e2aaf2c1cb - Merge pull request #5723 from dragos/issue/regression-assert-ide (11 days ago) <Lukas Rytz>
|\ \ \ \
| |/ / /
|/| | |
| * | | e5c957e764 - Fix regression in 5751763 (12 days ago) <Iulian Dragos>
* | | |   f174bfb1b8 - Merge pull request #5731 from janekdb/issue/GH-644/fix-spec-latex-rendering (11 days ago) <Seth Tisue>
|\ \ \ \
| * | | | ba4c6d4d09 - GH-644: Remove static html styling of spec code blocks (11 days ago) <Janek Bogucki>
|/ / / /
* | | |   8e40bef831 - Merge pull request #5729 from scala/revert-5658-topic/hashhash (12 days ago) <Adriaan Moors>
|\ \ \ \
| |_|/ /
|/| | |
| * | | 86cd70f396 - (origin/revert-5658-topic/hashhash) Revert "Fix erasure of the qualifier of ##" (12 days ago) <Adriaan Moors>
|/ / /
* | |   cbf7daa57d - Merge pull request #5681 from Philippus/issue/9704 (13 days ago) <Lukas Rytz>
|\ \ \
| * | | b8a8ac1556 - moved Pattern and TagsNotToClose to a HtmlTag companion object (13 days ago) <Philippus Baalman>
| * | | a019082bc3 - SI-9704 don't add a closed HtmlTag if it is already closed (4 weeks ago) <Philippus Baalman>
|  / /
* | |   effde0cd2c - Merge pull request #5726 from scala/revert-5629-issue/10120-quote-err (13 days ago) <Adriaan Moors>
|\ \ \
| * | | d60f6e3407 - (origin/revert-5629-issue/10120-quote-err) Revert "SI-10133 Require escaped single quote char lit" (13 days ago) <Adriaan Moors>
|/ / /
* | |   a8c4a54d18 - Merge pull request #5663 from gourlaysama/ticket/sd-256-enable-repl-colors-unix-2 (13 days ago) <Adriaan Moors>
|\ \ \
| * | | 6411170ede - SD-256 enable colored output by default on unix (13 days ago) <Antoine Gourlay>
* | | |   c96a977989 - Merge pull request #5658 from retronym/topic/hashhash (13 days ago) <Lukas Rytz>
|\ \ \ \
| |/ / /
|/| | |
| * | | f85c62e33e - Fix erasure of the qualifier of ## (6 weeks ago) <Jason Zaugg>
|  / /
* | |   76bfb9e0e5 - Merge pull request #5708 from szeiger/issue/si10194 (13 days ago) <Lukas Rytz>
|\ \ \
| * | | 1d22ee4ba3 - SI-10194: Fix abstract type resolution for overloaded HOFs (13 days ago) <Stefan Zeiger>
|  / /
* | |   dabec1a262 - Merge pull request #5700 from retronym/ticket/10154-refactor (13 days ago) <Lukas Rytz>
|\ \ \
| * | | 06eee798f5 - Refactor implementation of lookupCompanion (2 weeks ago) <Jason Zaugg>
|  / /
* | |   2f1e0c22ec - Merge pull request #5704 from som-snytt/issue/10190-elide-string (13 days ago) <Lukas Rytz>
|\ \ \
| * | | 6fb382501f - SI-10190 Elide string to empty instead of null (3 weeks ago) <Som Snytt>
|  / /
* | |   13f7b2a975 - Merge pull request #5640 from optimizely/repl-import-handler (2 weeks ago) <Adriaan Moors>
|\ \ \
| * | | aa7e3359be - Fix ImportHandler's reporting of importedNames and importedSymbols (8 weeks ago) <Hao Xia>
| * | | c89d821a4f - Fix SIOOBE in Name#pos for substrings of length 1 (8 weeks ago) <Jason Zaugg>
|  / /
* | |   023a96afe3 - Merge pull request #5629 from som-snytt/issue/10120-quote-err (2 weeks ago) <Adriaan Moors>
|\ \ \
| * | | 05cc3e2271 - SI-10120 ReplReporter handles message indent (7 weeks ago) <Som Snytt>
| * | | 939abf1c70 - SI-10120 Extra advice on unclosed char literal (8 weeks ago) <Som Snytt>
| * | | 855492c928 - SI-10133 Require escaped single quote char lit (8 weeks ago) <Som Snytt>
|  / /
* | |   e21ab42588 - Merge pull request #5660 from som-snytt/issue/9464-spec (2 weeks ago) <Adriaan Moors>
|\ \ \
| * | | a6dccebef2 - SI-9464 Clarify spec on no final trait (6 weeks ago) <Som Snytt>
|  / /
* | |   e87a43615d - Merge pull request #5659 from retronym/ticket/10026 (2 weeks ago) <Adriaan Moors>
|\ \ \
| |/ /
|/| |
| * | 777a0e5ae3 - SI-10026 Fix endless cycle in runtime reflection (2 weeks ago) <Jason Zaugg>
| |/
* |   23e5ed9219 - Merge pull request #5707 from retronym/topic/java9-prepare (2 weeks ago) <Lukas Rytz>
|\ \
| * | 96e8e9789b - Workaround bug in Scala runtime reflection on JDK 9 (3 weeks ago) <Jason Zaugg>
| * | fe2d9a431a - Avoid ambiguous overload on JDK 9 (3 weeks ago) <Jason Zaugg>
| * | 6bba8f7168 - Adapt to change in ClassLoader in JDK 9 (3 weeks ago) <Jason Zaugg>
| * | 8136057e7c - Bump scala-asm version (3 weeks ago) <Jason Zaugg>
|  /
* |   cad3c3da3e - Merge pull request #5709 from adriaanm/sam_wild_bound (2 weeks ago) <Lukas Rytz>
|\ \
| * | c396e965f1 - Ignore BoundedWildcardType in erasure type map (2 weeks ago) <Adriaan Moors>
* | |   3e9df417ed - Merge pull request #5711 from retronym/ticket/jrt (2 weeks ago) <Lukas Rytz>
|\ \ \
| * | | 09c7edc8a8 - Faster and simpler Java 9 classpath implementation (2 weeks ago) <Jason Zaugg>
|  / /
* | |   7b9d3cce97 - Merge pull request #5713 from janekdb/issue/GH-644/sync-jekyll-README-to-Gemfile (2 weeks ago) <Lukas Rytz>
|\ \ \
| * | | 5e5ec9aa40 - GH-644: Expand note regarding Jekyll versions (2 weeks ago) <Janek Bogucki>
|  / /
* | |   144f7e0097 - Merge pull request #5714 from dragos/issue/usage-sterr-SI-10178 (2 weeks ago) <Lukas Rytz>
|\ \ \
| |/ /
|/| |
| * | 640c85e08a - SI-10178 Route reporter.echo to stdout (2 weeks ago) <Iulian Dragos>
|  /
* |   2fec08b02a - Merge pull request #5717 from som-snytt/issue/10148-followup (2 weeks ago) <Adriaan Moors>
|\ \
| |/
|/|
| * f3d271b514 - SI-10148 Accept verbose zero (2 weeks ago) <Som Snytt>
* 147e5dd1b8 - Merge pull request #5716 from adriaanm/i296 (2 weeks ago) <Jason Zaugg>
* 12437a06d8 - Ensure ordering for args to `choose` in DurationTest (2 weeks ago) <Adriaan Moors>

SethTisue committed:
Merge pull request #5759 from SethTisue/merge-2.12.x-onto-2.13.x-20170306

[ci: last-only] merge 2.12.x onto 2.13.x (March 6, 2017)

lrytz committed:
Merge branch '2.12.x' into 2.13.x

lrytz committed:
Merge pull request #5763 from lrytz/merge-2.12-to-2.13-mar-8

Merge 2.12 to 2.13

lrytz committed:
Merge remote-tracking branch 'upstream/2.12.x' into merge-2.12-to-2.13-mar-9

lrytz committed:
Merge pull request #5768 from lrytz/merge-2.12-to-2.13-mar-9

Merge 2.12 to 2.13 mar 9

som-snytt committed:
SI-9745 typedFunction keeps x => f(x)

To type `x => f(x)`, `typedFunction` uses the expected
type and `f` to type `x`. Previously, it discarded the
result of typechecking `f`, which was a problem in the
case of a `g(e)` where `e` is "erroneous" but rewriteable
to an expr that checks, e.g., `g(i += 1)`.

This commit keeps the rewritten `g(i = i + 1)` instead
of trying to recover the erroneous tree.

som-snytt commented:
Two tests to add:

```
  // pos, crashes now
  def f(i: Int)(z: Any): Int = ???
  val g = (x: Int, y: Int) => f(42)(x, y)

  // neg, adds arity error which is currently silent
  def f(i: Int)(z: Any): Int = ???
  val g = (x, y) => f(42)(x, y)
```

lrytz commented on src/compiler/scala/tools/nsc/typechecker/Typers.scala:
```diff
@@ -254,7 +254,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
-    // for use with silent type checking to when we can't have results with undetermined type params
+    // for use with silent type checking for when we can't have results with undetermined type params
     // note that this captures the context var
     val isMonoContext = (_: Any) => context.undetparams.isEmpty
```
unused now

som-snytt replied:
yup.

lrytz commented on test/files/neg/t9745.check:
```diff
@@ -0,0 +1,27 @@
+ found   : (z: Any)Int
+ required: (x$1: ?, x$2: ?)?
+  val g = (x, y) => f(42)(x, y)
+                     ^
```
This one is a bit unfortunate..

som-snytt replied:
I'm on the fence.

lrytz reviewed (commented):
The new additional error is unfortunate, but I don't see a way out. Is it right that this version doesn't fix more / other issues than #5187, but just avoids type-checking the same tree twice?

lrytz committed:
Merge pull request #5358 from som-snytt/issue/9380

SI-9380 Remove Regex.unapplySeq(Any)

lrytz committed:
Merge pull request #5595 from monkey-mas/remove-unnecessary-methods-from-traversableonce

Remove unnecessary methods in TraversableOnce

SethTisue committed:
Merge pull request #5603 from szeiger/wip/modularize-parallel-collections

[ci:last-only] Remove parallel collections from scala-library

SethTisue committed:
remove/privatize deprecated stuff from collection.concurrent.TrieMap

so we can have a deprecation-free parallel collections module

SethTisue committed:
unbundle scala-library-all, scala-swing, scala-parser-combinators

SethTisue committed:
make UnPickler advise you of missing scala-parallel-collections.jar

janekdb committed:
Respell elidable checking function and make more similar to namesake

The layout and message edits bring out the similarity of the methods.

SethTisue committed:
Merge branch '2.12.x' into 2.13.x

no interesting merge conflicts.  versions.properties is always a
conflict factory, and HashMap.scala was because of parallel
collections removal.

% export mb=$(git merge-base origin/2.12.x origin/2.13.x)
% git log --graph --oneline --decorate $mb..origin/2.12.x | cat
*   96cdfb4403 (origin/HEAD, origin/2.12.x) Merge pull request #5741 from monkey-mas/bump-up-sbt-jmh-to-0.2.21
|\
| * 08f753e953 Bump up sbt-jmh to 0.2.21
*   38baf2ba9b Merge pull request #5771 from som-snytt/issue/regex-doc-err
|\
| * cc1b8f37a4 Fix and improve Regex doc
* |   6f905fd573 Merge pull request #5747 from dwijnand/fix-root-package-task
|\ \
| * | 4c1c8de34b Rewire package to osgiBundle for OSGi-enabled projects
|  /
* |   57948c8745 Merge pull request #5791 from SethTisue/cruft-begone
|\ \
| * | 4b9816100c remove test/pending directory too
| * | 25048bc737 (SethTisue/cruft-begone) rm -r test/{flaky,disabled*,checker-tests,support,debug}
| * | 0563c4b23c remove orphaned checkfiles
| * | 36357f3f83 remove two empty source files
| * | 8690809333 fix typos
| * | 8177b99fa6 unset a stray execute bit
* | |   c7c21525ec Merge pull request #5776 from lrytz/numbersCleanup
|\ \ \
| * | | 77e0602483 cleanups and clarifications in versions.properties
|  / /
* | |   95a263ecec Merge pull request #5789 from ceeph/patch-1
|\ \ \
| |/ /
|/| |
| * | a9ba3e675a Fix table formatting
|/ /
* |   6048c661f7 Merge pull request #5780 from lrytz/bootstrapOverwrite
|\ \
| * | 6ff389166e Use a single repository in the bootstrap job
|/ /
* |   0dab108872 Merge pull request #5777 from som-snytt/issue/10226
|\ \
| * | c77b420769 SI-10226 REPL handles paste when colorized
|/ /
* |   d0c26206e6 Merge pull request #5756 from rorygraves/2.12.x_depthboxing
|\ \
| * | f7cd76b532 Reduce boxing of scala.reflect.internal.Depth
|  /
* |   17e057bcc8 Merge pull request #5755 from rorygraves/2.12.x_map4
|\ \
| * | 0c6d4e5972 Performance improvements for Map4 to HashMap nad Set4 to HashSet transitions
| * | aefc8f4936 Add benchmarks for Map4 to HashMap and Set4 to HashSet transitions
| * | fad8b95335 Fix compile error on existing ListBenchmark
|  /
* |   3fadf69917 Merge pull request #5675 from piyush-jaiswal/issue/9729
|\ \
| |/
|/|
| * cc99026540 Add tests for ConsoleReporter.
*   6e9268b08e Merge pull request #5761 from lrytz/sd329
|\
| * 6abb6ba47a Don't use `equals` for comparing java.lang.Double/Float
*   680d866142 Merge pull request #5719 from retronym/ticket/10187
|\
| * 898fa00a76 SI-10187 Support mutation of mutable.HashMap in getOrElseUpdate
*   0e38f28e33 Merge pull request #5767 from som-snytt/issue/5621
|\
| * dea2171a1c SI-5621 Missing implicits are supplied by defaults
*   482879f27b Merge pull request #5769 from som-snytt/issue/8969
|\
| * 050811bd84 SI-8969 Accept poly+implicit for assignment syntax
* cba77cece5 Merge pull request #5766 from lrytz/versionsReadme
* 8fcb9f170e Adapt README to new version numbers
% git merge origin/2.12.x
% git status
...
	both modified:   src/library/scala/collection/mutable/HashMap.scala
	deleted by them: test/pending/shootout/fasta.scala
	both modified:   versions.properties
% git rm --force test/pending/shootout/fasta.scala
% emacs versions.properties
% git add -u versions.properties
% emacs src/library/scala/collection/mutable/HashMap.scala
% git add -u src/library/scala/collection/mutable/HashMap.scala

lrytz committed:
Merge pull request #5794 from SethTisue/merge-2.12-to-2.13-mar-21

Merge branch '2.12.x' into 2.13.x

janekdb committed:
Remove unused 2.8.0 Scala version val

lrytz committed:
Merge pull request #5644 from xuwei-k/MonadOps-value-class

Make TraversableOnce.MonadOps a value class

lrytz committed:
Merge pull request #5788 from SethTisue/triemap-deprecations

remove/privatize deprecated stuff from collection.concurrent.TrieMap

lrytz committed:
Merge pull request #5793 from janekdb/topic/2.13/elisible-elidable

Respell elidable checking function and make more similar to namesake

lrytz committed:
Merge pull request #5795 from janekdb/topic/2.13/remove-280-constant

Remove unused 2.8.0 Scala version val

lrytz committed:
Merge pull request #5611 from fhalde/2.12.x

Common usage of finding random number in a range [ci: last-only]

lrytz committed:
Set scala.binary.version to 2.12 until 2.13.0-M1 is out

The current override in VersionUtil hard-codes the binary version
to `2.12`, which means that the `-Dscala.binary.version` override
in the bootstrap script does not have its intended effect. So we
were in fact using the 2.12 binary modules also when testing the
boostrapped 2.13 builds.

SethTisue committed:
Merge pull request #5810 from lrytz/binaryVersion

Set scala.binary.version to 2.12 until 2.13.0-M1 is out

lrytz committed:
Avoid using code that will move to modules

`scala.concurrent`, `scala.Console`, `scala.compat._`,
`sys.runtime`, `sys.addShutdownHook`, `sys.allThreads`,
`sys.env`.

Also fixes a test reducing references to static modules (Console, Predef)
delays initialization of object `NoStackTrace`. In `t2318`, this may now
happen *after* the security manager is installed. Therefore the security
manager needs to allow reading the system property
`scala.control.noTraceSuppression`

adriaanm committed:
Merge pull request #5698 from szeiger/wip/overloaded-pf-inference

Eliminate non-matching overloads early for partial function literals

lrytz committed:
Replace mkSysErrorCall by mkThrowNewRuntimeException

Since we no longer want to depend on `sys.error`, the compiler
also shouldn't generate calls to it.

This also fixes the OSGI tests. The failure was interesting: once
there were no more refernces to `scala.sys` left in scala-reflect,
the dependency on that package disappeared from scala-reflect.jar's
MANIFEST.MF. When instantiating scala.reflect.runtime.universe through
an osgi classloader, `getPackageObject("scala.sys")` in definitions
failed becasue the package could not be found through that classloader.

lrytz committed:
Merge pull request #5790 from SethTisue/no-more-scala-library-all

unbundle scala-library-all, scala-swing, scala-parser-combinators

adriaanm committed:
Bye bye JIRA

lrytz committed:
Merge pull request #5830 from lrytz/cut-deps

Cut dependencies on future modules

cypok committed:
Stricter type parameters of HashEntry

lrytz committed:
Merge commit '96cdfb4' into 2.13.x

lrytz committed:
Merge remote-tracking branch 'upstream/2.12.x' into 2.13.x

SethTisue committed:
Merge pull request #5843 from lrytz/merge-2.12-to-2.13-apr-11

Merge 2.12 to 2.13 [ci: last-only]

lrytz committed:
Use partest 1.1.1

During bootstrapping 2.13.x, partest is re-built. Partest has Xlint
fatal warnings enabled.

scala/scala#5402 enabled new unused warnings, which fails the 1.1.0
build.

lrytz committed:
Merge pull request #5849 from lrytz/partest-1.1.1

Use partest 1.1.1

lrytz committed:
Merge remote-tracking branch 'upstream/2.12.x' into merge-2.12-to-2.13-apr-13

lrytz committed:
Merge pull request #5850 from lrytz/merge-2.12-to-2.13-apr-13

Merge 2.12 to 2.13

lrytz committed:
Bump version numbers after 2.13.0-M1

lrytz committed:
Merge pull request #5858 from lrytz/versionsBump13M1

Bump version numbers after 2.13.0-M1

lrytz committed:
Use 2.13 binary version for scalacheck

lrytz committed:
Merge pull request #5860 from lrytz/scalacheckUpdate

Use 2.13 binary version for scalacheck

SethTisue committed:
Merge remote-tracking branch 'origin/2.12.x' into merge-2.12-to-2.13-apr-24

trivial conflicts only

lrytz committed:
Merge pull request #5862 from SethTisue/merge-2.12-to-2.13-apr-24

Merge 2.12.x to 2.13.x [ci: last-only]

lrytz committed:
Merge remote-tracking branch 'upstream/2.12.x' into 2.13.x

lrytz committed:
Remove unnecessary abstraction

Added in ced7214959, no longer needed since ICodeReader is gone.

SethTisue committed:
Merge pull request #5868 from lrytz/merge-2.12-to-2.13-apr-27

Merge 2.12 to 2.13

lrytz committed:
Invalidate symbols for artifact classfiles, refactor classfile parser

No longer run the classfile parser on Scala generated classfiles that
don't have a Scala signature (module classes, inner classes, etc).

Various cleanups in the classfile parser, minimize the work performed
on Scala classfiles. Before, the attributes section was parsed twice:
once to find the ScalaSig attribute, the second time to find the
ScalaSignature in the RuntimeVisibleAnnotations. Now everything happens
in the first iteration.

Also fixes a bug in the backend: classes ending in `$` did not get a
ScalaSignature by mistake. They were filtered out by the name-based
test that is supposed to identify module classes.

adriaanm committed:
Small cleanups in the REPL, preparing modularization

SethTisue committed:
Merge remote-tracking branch 'origin/2.12.x' into merge-2.12-to-2.13-may-1

SethTisue committed:
Merge pull request #5878 from SethTisue/merge-2.12-to-2.13-may-1

Merge 2.12 to 2.13 [ci: last-only]

adriaanm committed:
Merge branch '2.12.x' into 2.13.x

adriaanm committed:
Split REPL into front / back; drop embedded jline.

Backend is like presentation compiler: API offered by compiler.
Frontend is the shell, which uses jline. Should be easier
to implement different frontends (ammonite/spark worksheets).

Hopefully the modularization will eliminate the need for the
shaded copy of jline.

Remove some deprecated methods, but leave minimal stubs in
place for sbt 0.13...

Adjust cursor in repl completion for partial input and
leading dot completion; caught by CompletionTest.

adriaanm committed:
Deeper simplifications, correct error line number

Simplify splash loop, drop splash reader;
explicitly initialize completion.

Instead of code, indent messages in reporter.
Better line numbers in error/warning messages,
enabled by refactoring in request compilation.

lrytz committed:
Merge pull request #5902 from adriaanm/2.13.x

Merge 2.12.x into 2.13.x [ci: last-only]

adriaanm committed:
Build: command parser should include separators

adriaanm committed:
In-source essence of sbt console interface

adriaanm committed:
Reduce mutation, single-use abstractions

Also accept tab completion during startup,
as soon as the repl has initialized.

adriaanm committed:
Use NIO in FileBackedHistory, drop dependency on repldbg

Behavior change: fail if neither properties "scala.shell.histfile",
or "user.home" are set, rather than creating a history file in the
file system root.

adriaanm committed:
Split Scripted in frontend/repl

adriaanm committed:
Separate shell from compiler, reduce mutation

The shell (ILoop) should only depend on the repl (IMain),
and not on the rest of the compiler. Working towards a
stable API that multiple frontends could use to provide
a shell on top of a repl-oriented compiler (in the same
vein as IDEs built on the presentation compiler).

Move config/settings out of package object, and into
ShellConfig / ReplReporterConfig.

IMain should not do any reporting to the console,
just send strings / trees to frontend, which will
then send it to output (colorizing where required).

TODO:
  - bring back color output by parsing trees received
    by shell from imain

Misc:
  - Rename unwrap to rootCause
  - Direct way to make welcome message reproducible for tests
  - paste prints a bit more prompt

adriaanm committed:
Reduce ReplReporter interface

Simplify compiler initialization

adriaanm committed:
Move compiler-coupled commands to IMain

Reduce the IMain interface used by the shell.
Should capture this in an actual trait.

So long, IMainOps

adriaanm committed:
Restrict shell's dependency on `IMain` to `Repl`

Carving out an interface that could cross process boundaries.
(Strings/Trees/Positions, but no types/symbols)

Similarly, model the subset of the repl used by sbt as `ReplCore`.
For sbt/zinc 1.0, we should really sit down and agree on a new
interface.

retronym committed:
Merge remote-tracking branch 'origin/2.12.x' into merge/2.12.x-to-2.13.x-20170530

Conflicts:
   versions.properties

Conflict related to the unbundling of scala-swing, scala-parser-combinators in
2.13.x, trivially resolved.

SethTisue committed:
Merge pull request #5922 from retronym/merge/2.12.x-to-2.13.x-20170530

Merge 2.12.x to 2.13.x [ci: last-only]

szeiger committed:
Merge pull request #5616 from monkey-mas/change-signature-of-WrappedArray-newBuilder

Change the signature of WrappedArray.newBuilder[T] to return WrappedArrayBuilder[T]

szeiger committed:
Merge pull request #5840 from cypok/hashentry-stricter-type

Stricter type parameters of collection.mutable.HashEntry

adriaanm committed:
Review feedback from lrytz and retronym

adriaanm committed:
Drop welcome message from repl test output

adriaanm committed:
Exit repl on first CTRL-d

xuwei-k committed:
Privatize underlying values in value class

adriaanm committed:
Improve repl error positions, tree gen

Parse user input as-is, and use that as content of
unit's source file content. The unit's body has the
trees resulting from wrapping and rewriting the user's
input to implement the repl's semantics (store the
result in a val etc).

Also pass on whether parsed trees had xml elements.

This allows us to be more precise in our error messages,
as positions map 1-1 to user input (as suggested by retronym,
who helped getting the range positions right).

Note: input may contain leading/trailing non-trees.
For example, whitespace/comments don't survive as trees,
thus we can't assume the trees cover the range pos
(0, buf.length); instead, compute it.

Goal: no string-based code manipulation. No there yet.

Handle assignment minimally to avoid memory leaks
So, don't hang on to assigned values.
Still provide some feedback that the assignment has
been executed --> `"mutated x"`.

Also, bring back pure expression warning to maintain
status quo -- but who wants to see those pure expr
warnings in the repl??

Rangepos invariants require most synthetic code
to have a transparent pos, but the presentation
compiler locates trees by position, which means
use code must have an opaque position, and we
must be able to navigate to it from the root,
which means all ancestor trees to user code must
cover the positions of said user code.

lrytz committed:
Merge pull request #5945 from xuwei-k/privatize

Privatize underlying values in value class

adriaanm committed:
wrappingPos considers positions of annotations

adriaanm committed:
Partial input only needed for completion

Move it there, and while we're at it, simplify Completion hierarchy

adriaanm committed:
Simplify completion hierarchy some more

lrytz committed:
ScalaSigBytes is no longer a ClassfileAnnotArg

This abstraction was not necessary, it was just dealing with encoding
the signature. This is now handled more directly.

Jasper-M committed:
Add apply methods to typeclass companion objects in scala.math

Add apply methods to companion objects of Numeric, Integral and Fractional.
Make the apply methods in the companion objects of Numeric, Intergral, Fractional, Equiv and Ordering @inline.

lrytz committed:
Merge pull request #5952 from lrytz/skipScalaRaw

Invalidate symbols for artifact classfiles, refactor classfile parser

adriaanm committed:
Merge pull request #5903 from adriaanm/repl-modularize

REPL: decouple the UI (shell) from the compiler [ci: last-only]

SethTisue committed:
continue supporting Exceptional.unwrap for now

since not having it is breaking building partest during bootstrap,
and it's not worth branching partest over

SethTisue committed:
Merge remote-tracking branch 'origin/2.12.x' into merge/2.12.x-to-2.13.x-20170626

Only an easy conflict in ILoop.scala

% git fetch origin --quiet

% git checkout -b merge/2.12.x-to-2.13.x-20170626 origin/2.13.x
Branch merge/2.12.x-to-2.13.x-20170626 set up to track remote branch 2.13.x from origin.
Switched to a new branch 'merge/2.12.x-to-2.13.x-20170626'

% export MB=$(git merge-base HEAD origin/2.12.x)

% git log --graph --oneline --decorate $MB..origin/2.12.x | gist
https://gist.github.com/73cf792b1b46f8ba8eeeabeb5f7e5f20

% git merge origin/2.12.x
Auto-merging versions.properties
Auto-merging src/scaladoc/scala/tools/nsc/doc/model/ModelFactory.scala
Auto-merging src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala
CONFLICT (content): Merge conflict in src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala
Auto-merging src/reflect/scala/reflect/internal/Types.scala
Auto-merging src/reflect/scala/reflect/internal/StdNames.scala
Auto-merging src/library/scala/collection/immutable/Queue.scala
Auto-merging src/library/scala/collection/TraversableLike.scala
Auto-merging src/compiler/scala/tools/nsc/typechecker/SuperAccessors.scala
Auto-merging src/compiler/scala/tools/nsc/transform/UnCurry.scala
Auto-merging src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala
Auto-merging src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala
Auto-merging src/compiler/scala/tools/nsc/Global.scala

SethTisue committed:
Merge pull request #5961 from SethTisue/fix-partest-bootstrap

continue supporting Exceptional.unwrap for now

SethTisue committed:
Merge pull request #5963 from SethTisue/merge/2.12.x-to-2.13.x-20170626

Merge 2.12.x to 2.13.x [ci: last-only]

retronym commented:
I'm well impressed that we've found the spot to fix this. But I'm hesitant to touch this area in 2.12.x. Let's target at 2.13.x and re-rereview. @som-snytt I'm happy to do the busywork if you've expended your exasperation budget on this one...

som-snytt commented:
Since I have no memory of this change, I'm willing to start from scratch. There's a previous comment where I was on the fence about something, I wonder if it matters which side one comes down on.

lrytz committed:
Merge pull request #5954 from Jasper-M/topic/summon-math-typeclasses

Add apply methods to typeclass companion objects in scala.math

szeiger committed:
Two fixes for overload resolution for PartialFunctions

The extension of shape-based overload resolutions for pattern-matching
anonymous functions in https://github.com/scala/scala/pull/5698 did not
work correctly in all cases:

- Type unification for shapes needs to treat `PartialFunction` types
  the same as `FunctionN` and SAM types. They were considered for
  unification but the parameter type was not extracted correctly.

- When the matched argument types are constructed after a successful
  shape match, a pattern-matching anonymous function has to be typed as
  `PartialFunction` instead of `FunctionN`.

NthPortal committed:
scala/bug#5961 Add Ordering composition methods

Add `Ordering.orElse(Ordering[T]):Ordering[T]` and
`Ordering.orElseBy[S:Ordering](f:T=>S):Ordering[T]`.

Fixes scala/bug#5961

szeiger committed:
Merge pull request #5523 from ruippeixotog/issue/9706

SI-9706 Add a fromString method to Numeric trait

szeiger committed:
Merge pull request #5623 from johnynek/oscar/liskov-leibiz

Add substitute methods to <:< =:= making these Liskov/Leibniz

szeiger committed:
Merge commit '65486a8423' into merge/2.12.x-to-2.13.x-20170717

szeiger committed:
Merge commit 'c4a732edab' into merge/2.12.x-to-2.13.x-20170717

szeiger committed:
Merge commit 'ae9b465f3d' into merge/2.12.x-to-2.13.x-20170717

szeiger committed:
Merge commit '79eeec3c9a' into merge/2.12.x-to-2.13.x-20170717

szeiger committed:
Merge commit 'a51b9865f5' into merge/2.12.x-to-2.13.x-20170717

szeiger committed:
Merge remote-tracking branch 'scala/2.12.x' into merge/2.12.x-to-2.13.x-20170717

szeiger committed:
Merge pull request #5951 from szeiger/issue/10081-2.13

Avoid diverging implicit scope search

szeiger committed:
Merge pull request #5988 from NthPortal/bug#5961/R3

scala/bug#5961 Add Ordering composition methods

szeiger committed:
Merge pull request #5997 from szeiger/merge/2.12.x-to-2.13.x-20170717

Merge/2.12.x to 2.13.x 20170717 [ci: last-only]

szeiger committed:
Don’t create PartialFunction unless necessary (or hard to avoid)

Simpler version of https://github.com/scala/scala/pull/5975/commits/3c95dac0dcbb0c8eb4686264026ad9c86b2022de
which doesn’t retype the PartialFunction argument as Function1 after a
mixed PartialFunction / Function1 overload was resolved to a Function1
alternative.

szeiger committed:
Merge commit '085be56899' into merge/2.12.x-to-2.13.x-2017020

szeiger committed:
Merge commit '0dc2c3dc60' into merge/2.12.x-to-2.13.x-2017020

szeiger committed:
Merge commit '71f9b43c9e' into merge/2.12.x-to-2.13.x-2017020

szeiger committed:
Merge commit '18721ead44' into merge/2.12.x-to-2.13.x-2017020

szeiger committed:
Merge remote-tracking branch 'scala/2.12.x' into merge/2.12.x-to-2.13.x-2017020

szeiger committed:
Merge pull request #6001 from szeiger/merge/2.12.x-to-2.13.x-2017020

Merge/2.12.x to 2.13.x 2017020 [ci: last-only]

szeiger committed:
Merge pull request #6003 from szeiger/fix/pf-overloads2

Two fixes for overload resolution for PartialFunctions

som-snytt commented:
Superseded https://github.com/scala/scala/pull/6007
